### PR TITLE
Feature/issue 15- DownloadScreen - Update UI

### LINKED ISSUE #15: [Feature] Download Screen/ Update UI
### Overview
- //  Update UI

### Impact
- // Home Screen

### Screenshots (Optional)
- // Capture affected screens
Simulator Screen Shot - iPhone 12 Pro - 2021-10-19 at 17 28 12
Simulator Screen Shot - iPhone 12 Pro - 2021-10-19 at 17 28 42

### PR DESCRIPTION
**Pull request by minh5896**
## Prerequisite
- [x] Self-review :monocle_face:
- [x] No Crashes ☠️
---

### Root cause
- // update UI 

### Solution
- // update UI

### Impact
- // Download Screen

### Screenshots (Optional)
- // Capture affected screens

Simulator Screen Shot - iPhone 12 Pro - 2021-10-19 at 17 28 12
Simulator Screen Shot - iPhone 12 Pro - 2021-10-19 at 17 28 42

### CODE REVIEW

File: MovieApp/Screens/Details/DetailsViewController.swift [hunk 1]
```diff
@@ -12,4 +12,16 @@ class DetailsViewController: BaseViewController {
 	override func viewDidLoad() {
 	super.viewDidLoad()
 	}
```
Comment (namnh92) [inline MovieApp/Screens/Details/DetailsViewController.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Code4Fun-Group/Movie-App/blob/5674ecfc29f32c9bbaa1d4d78e88396268dceeb4/MovieApp/Screens/Details/DetailsViewController.swift#L14"></span>Override methods which only call super can be removed</td>
    </tr>
  </tbody>
</table>

File: MovieApp/Screens/Home/Models/DownloadCell.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import UIKit
+
+class DownloadCell: UICollectionViewCell {
+// MARK: - Iboutlets
+	@IBOutlet private weak var imageDownloadCell: UIImageView!
+
+// MARK: - Life cycles
+	override func awakeFromNib() {
+		super.awakeFromNib()
+	}
```
Comment (namnh92) [inline MovieApp/Screens/Home/Models/DownloadCell.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Code4Fun-Group/Movie-App/blob/5674ecfc29f32c9bbaa1d4d78e88396268dceeb4/MovieApp/Screens/Home/Models/DownloadCell.swift#L17"></span>Override methods which only call super can be removed</td>
    </tr>
  </tbody>
</table>

File: MovieApp/Screens/Details/DetailsViewController.swift (hunk 1 above)
Comment (minh5896) [reply MovieApp/Screens/Details/DetailsViewController.swift]: fix compplete

File: MovieApp/Screens/Home/Models/DownloadCell.swift (hunk 2 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Models/DownloadCell.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 3]
```diff
@@ -7,11 +7,131 @@
...
+// MARK: - Life cycles
 	override func viewDidLoad() {
 		super.viewDidLoad()
-		self.view.backgroundColor = .black
-		
+		viewModel = DownloadViewModel(viewController: self)
+		downloadCollectionView.delegate = self
+		downloadCollectionView.dataSource = self
+		setupUI()
+		setupTableView()
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]:
```suggestion
```
@minh5896 : Move to setupUI()

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 4]
```diff
@@ -7,11 +7,131 @@
...
+	func showError() {
+		downloadCollectionView.isHidden = true
+		view.addSubview(imageDownloadView())
+		view.addSubview(downloadButton())
+	}
+}
+// MARK: - UICollectionViewDelegateFlowLayout
+extension DownloadViewController: UICollectionViewDelegateFlowLayout {
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+		return CGSize(width: collectionView.frame.width / 3, height: collectionView.frame.height / 4 )
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]: @minh5896 : Make constants

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 5]
```diff
@@ -7,11 +7,131 @@
...
+	}
+
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+		return 0
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]: @minh5896 : Make constants

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 6]
```diff
@@ -7,11 +7,131 @@
...
+	}
+
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+		return 0
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]: @minh5896 : Make constants

File: MovieApp/Screens/Download/ViewModels/DownloadViewModel.swift [hunk 7]
```diff
@@ -10,12 +10,30 @@ import Foundation
...
 extension DownloadViewModel: IDownloadViewModel {
 	func getMovies() {
+		DependencyResolver.shared.movieAPIService.getMovies { [weak self] result in
+			guard let self = self else { return }
+			switch result {
+			case .success(let movies):
+				self.downloadMovieViewModels = movies.compactMap({
+					DownloadMovieViewModel(item: $0)
+				})
+//			self.viewController?.showError()
```
Comment (namnh92) [inline MovieApp/Screens/Download/ViewModels/DownloadViewModel.swift]: @minh5896 : Remove comment
```suggestion
```

File: MovieApp/Screens/Home/Views/ContinueCell/ContinueCell.swift [hunk 8]
```diff
@@ -42,6 +42,10 @@ extension ContinueCell: UICollectionViewDataSource {
 	func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
 		guard let cell = self.continueCollectionView.dequeueReusableCell(withReuseIdentifier: ConstantsCell.continueItemCell, for: indexPath) as? ContinueItemCell else { return UICollectionViewCell() }
 		cell.configure(with: continueData[indexPath.row])
+		cell.layer.masksToBounds = true
+		cell.layer.cornerRadius = 5
+		cell.layer.borderWidth = 2
+		cell.layer.borderColor = UIColor.white.cgColor
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/ContinueCell/ContinueCell.swift]: @minh5896 : Move configure layout to Cell

File: MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift [hunk 9]
```diff
@@ -13,7 +13,7 @@ class ListMovieCell: UITableViewCell {
 	@IBOutlet private weak var collectionListView: UICollectionView!
 
 // MARK: - Variables
-	 weak var delegate: PreviewCellDelegate?
+	 weak var delegate: CellDelegate?
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift]:
```suggestion
	weak var delegate: CellDelegate?
```

File: MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift [hunk 10]
```diff
@@ -44,6 +44,10 @@ extension ListMovieCell: UICollectionViewDataSource {
 	func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
 		guard let cell = self.collectionListView.dequeueReusableCell(withReuseIdentifier: ConstantsCell.listItemCell, for: indexPath) as? ListItemCell else { return UICollectionViewCell() }
 		cell.configure(with: listData[indexPath.row])
+		cell.layer.masksToBounds = true
+		cell.layer.cornerRadius = 5
+		cell.layer.borderWidth = 2
+		cell.layer.borderColor = UIColor.white.cgColor
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift]: @minh5896 : Move configure to Cell

File: MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift [hunk 11]
```diff
@@ -62,3 +66,10 @@ extension ListMovieCell: UICollectionViewDelegateFlowLayout {
 		return 5
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift]: @minh5896 : Make constants

**Review by namnh92 — CHANGES_REQUESTED** [review]

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 3 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 4 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 6 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

File: MovieApp/Screens/Download/ViewModels/DownloadViewModel.swift (hunk 7 above)
Comment (minh5896) [reply MovieApp/Screens/Download/ViewModels/DownloadViewModel.swift]: fix complete

File: MovieApp/Screens/Home/Views/ContinueCell/ContinueCell.swift (hunk 8 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/ContinueCell/ContinueCell.swift]: fix complete

File: MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift (hunk 10 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift]: fix complete

File: MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift (hunk 11 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift]: fix complete

File: MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift (hunk 9 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 5 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

**Comment by namnh92** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**